packages/engine/Source/Core/addAll.js

```diff
@@ -0,0 +1,41 @@
+import defined from "./defined.js";
+
+/**
+ * Adds all elements from the given source array to the given target array.
+ *
+ * If the `source` is `undefined`, then nothing will be done. Otherwise,
+ * this has the same semantics as
+ * ```
+ * for (const s of source) target.push(s);
+ * ```
+ * but is usually more efficient than a `for`-loop, and does not put the
+ * elements of the source on the stack, as it would be done with the
+ * spread operator or when using `target.push.apply(source)`.
+ *
+ * @function
+ * @private
+ *
+ * @param {Array|undefined} source The source array
+ * @param {Array} target The target array
+ *
+ * @example
+ * const target = [ 0, 1, 2 ];
+ * const source = [ 3, 4, 5 ];
+ * Cesium.addAll(source, target);
+ * // The target is now [ 0, 1, 2, 3, 4, 5 ]
+ */
+function addAll(source, target) {
+  if (!defined(source)) {
+    return;
+  }
+  const s = source.length;
+  if (s === 0) {
```

Comment:
it's a nitpick but in general I think we should avoid single letter variable names outside the looping variables `i`, `j`, `k`, and sometimes `l`.

Maybe it's just personal preference but I find this a little more readable even if it is more verbose
```js
  const sourceLength = source.length;
  if (sourceLength === 0) {
    return;
  }
  const targetLength = target.length;
  target.length += sourceLength;
  for (let i = 0; i < sourceLength; i++) {
    target[targetLength + i] = source[i];
  }
```

Reply:
I nearly espected that, and can change it accordingly. 
But regarding the variable name `1`: Here is an animated GIF of `l` and `1`, switching every second:

![Cesium L](https://github.com/user-attachments/assets/13952899-63b1-406b-8e7b-329e08968d98)

Reply:
Renamed to `sourceLength` and `targetLength`